databroker/src/grpc/kuksa_val_v1/val.rs

```diff
@@ -743,4 +898,100 @@ mod tests {
+
+        authorized_access
+            .add_entry(
+                "Vehicle.Speed".to_owned(),
+                broker::DataType::Float,
+                broker::ChangeType::OnChange,
+                broker::EntryType::Sensor,
+                "Test datapoint 1".to_owned(),
+                None,
+                Some("km/h".to_owned()),
+            )
+            .await
+            .expect("Register datapoint should succeed");
+
+        let streamed_update_request = proto::StreamedUpdateRequest {
+            updates: vec![proto::EntryUpdate {
+                fields: vec![proto::Field::Value as i32],
+                entry: Some(proto::DataEntry {
+                    path: "Vehicle.Speed".to_owned(),
+                    value: Some(proto::Datapoint {
+                        timestamp: Some(std::time::SystemTime::now().into()),
+                        value: Some(proto::datapoint::Value::Float(120.0)),
+                    }),
+                    metadata: None,
+                    actuator_target: None,
+                }),
+            }],
+        };
+
+        let mut streaming_request = tonic_mock::streaming_request(vec![streamed_update_request]);
+        streaming_request
+            .extensions_mut()
+            .insert(permissions::ALLOW_ALL.clone());
+        match broker.streamed_update(streaming_request).await {
+            Ok(response) => {
+                tokio::spawn(async move {
+                    let stream = response.into_inner();
+                    let mut receiver = stream.into_inner();
+                    let option = receiver.recv();
+                    assert!(option.await.is_none()) // no errors should occur and no ack is delivered
+                });
+            }
+            Err(_) => {
+                panic!("Should not happen")
+            }
+        }
+    }
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+    async fn test_streamed_update_with_invalid_datapoint() {
+        let broker = DataBroker::default();
+
+        let streamed_update_request = proto::StreamedUpdateRequest {
+            updates: vec![proto::EntryUpdate {
+                fields: vec![proto::Field::Value as i32],
+                entry: Some(proto::DataEntry {
+                    path: "Vehicle.Invalid.Speed".to_owned(),
+                    value: Some(proto::Datapoint {
+                        timestamp: Some(std::time::SystemTime::now().into()),
+                        value: Some(proto::datapoint::Value::Float(120.0)),
+                    }),
+                    metadata: None,
+                    actuator_target: None,
+                }),
+            }],
+        };
+
+        let mut streaming_request = tonic_mock::streaming_request(vec![streamed_update_request]);
+        streaming_request
+            .extensions_mut()
+            .insert(permissions::ALLOW_ALL.clone());
+        match broker.streamed_update(streaming_request).await {
```

Comment:
Calling broker.streamed_update requires a type Request<Streaming<StreamedUpdateRequest>>. However, after fidding around for hours I was not able to get this type without the help of the tonic_mock library. The tonic_mock library however forces us to update some lib versions (e.g. tonic, prost). Updating them introduces a lot of deprecation warnings and makes a lot of test fail: https://github.com/eclipse-kuksa/kuksa-databroker/actions/runs/9286289961/job/25552663996?pr=34

@argerus  Do you have an easy solution on how to create a "Request<Streaming<StreamedUpdateRequest>>"? 

Already tested stuff like this: 
- https://github.com/tyrchen/tonic-mock/pull/1#issuecomment-1241164173 (this basically should show an alternative solution on how we don't have to use tonic_mock but it might be to old by now?)
- https://www.qovery.com/blog/a-guided-tour-of-streams-in-rust/
- https://github.com/hyperium/tonic/blob/master/examples/routeguide-tutorial.md
- https://github.com/hyperium/tonic/blob/master/examples/src/streaming/server.rs#L97 (following the lines where it is called leads here: https://github.com/hyperium/tonic/blob/master/examples/src/streaming/client.rs#L11 but re-implementing this method for StreamedUpdateRequest does not work - it's simply not recognized by broker.streamed_update) 

None of them worked unfortunately and at this point in time I'm absolutely clueless what else to test

Reply:
`tonic_mock::streaming_request(...)` seems to have been created specifically for making this task easier, correct?

If so, I would suggest using it. If I understand you correctly, the issue is then that `::from_i32(...)` has been deprecated in newer versions of tonic.

I pushed a [branch](https://github.com/boschglobal/kuksa-databroker/tree/topic/fix_deprecated_from_i32) based on this PR that changes the use of this deprecated function into using the `TryFrom` trait instead.

Reply:
@argerus Thanks for your effort! I merged your branch into VAL-1320 and pushed the changes. 

Unfortunately that did not fix the failing tests. Maybe the issue comes from the cucumber setup but I am not able to debug it, making finding the root cause extra hard. I tried upgrading the version of cucumber to 0.21 but that did not fix it.

Reply:
Nevermind, I found the error. It seems to no longer possible to connect to databroker via tls if not explicitely using "https://". Also "grpc://" is no longer supported. I think I saw the use of "grpc://" in some places, so adaptions to documentation might be necessary?

The following link helped me: https://github.com/hyper*ium/tonic/issues/1496

Reply:
That line in quickstart is unrelated, as it is using python code (grpc(s) an ws(s) scheme is a convention in the python CLI to quickly differnatiate  between  websocket based VISS or GRPC based kuksa protocol with or without TLS)

Reply:
If the tonic behavior changed, do we need some related change in rust  cli or does that anyway only ever forward URIs given by the user?

Reply:
Well, good question, the databroker-cli as a default value uses `http://127.0.0.1:55555` (https://github.com/eclipse-kuksa/kuksa-databroker/blob/main/databroker-cli/src/cli.rs#L40). 

However as a default the tls options are disabled. So you could argument and say it's fine, but we probably should document somewhere, that when using `https` the url needs to be explicitely specified using `https://`, maybe once in the corresponding tls section and maybe once in some error faq, where you can look up the error and see a fix for it?

Reply:
I made some changes to the documentation with the latest commit and added some info about it, please have a look at it.

Reply:
Documentation update  good. Whether this PR techically "sound"/good enough I can not say